pubsub/pstest/fake.go

```diff
@@ -1074,6 +1113,33 @@ func (s *subscription) deleteStream(st *stream) {
 		s.streams = deleteStreamAt(s.streams, i)
 	}
 }
+
+func (s *subscription) deadLetterCandidate(m *message) bool {
+	if s.proto.DeadLetterPolicy == nil {
+		return false
+	}
+	if m.retriesDone(s.proto.DeadLetterPolicy.MaxDeliveryAttempts) {
+		return true
+	}
+	return false
+}
+
+func (s *subscription) publishToDeadLetter(m *message) {
+	acks := 0
+	if m.acks != nil {
+		acks = *m.acks
+	}
+	deliveries := 0
+	if m.deliveries != nil {
+		deliveries = *m.deliveries
+	}
+	s.deadLetterTopic.publish(m.proto.Message, &Message{
```

Comment:
Here I doubt that I need to use the original message properties (`deliveries` and `acks`) since the `publish` method is not allowing the proto values as well for these properties.
This is about the changes that suggested around `m.proto.DeliveryAttempt`.
Also you can check [fake_test.go#382](https://github.com/googleapis/google-cloud-go/pull/5320/files#diff-c6a74920cef99f2ab6281118b9eb61b39c99f9b17ee2961758db6688e0c6af9eR382)